internal/compiler/generator/cpp.rs

```diff
@@ -1953,7 +1953,10 @@ fn generate_public_api_for_properties(
             }));
             declarations.push(Declaration::Function(Function {
                 name: format!("on_{}", ident(&p.name)),
-                template_parameters: Some("typename Functor".into()),
+                template_parameters: Some(format!(
+                    "std::invocable<{}> Functor",
```

Comment:
This doesn't check the return value.
Adding a check for the return value might need some more changes to the Declaration::Function to add the requires close.
Unless we write our own concept

Reply:
I wonder which gives a better error message to users, our concept on a more complex expression in requires().